### ISSUE TITLE
logrotate configuration

### ISSUE THREAD

**Issue by jriguera** [go-graphite/go-carbon#67]
I have created a logrotate config file to be included in the deb and rpm packages.

The code is here:
https://github.com/jriguera/go-carbon/commit/4030c0dc07d715a98ce6435fc9b174c4a1f6fc5c

I think could be interesting to be included in the upstream repo via PR. Is it ok? or do you have other ideas of doing it?

Thanks!

**Comment by tehlers320** [comment]
Dont you want copytruncate on these?

**Comment by jriguera** [comment]
Yes, you are right! I will update the config

**Comment by lomik** [comment]
I do not think that logrotate configuration must come with the rpm. May be as sample in conf-examples directory

PS. `copytruncate` not required

**Comment by bzed** [comment]
#112 sounds like ```copytruncate``` is needed? The logrotate config should be shipped in the Debian package, not sure how CentOS handles it.

**Comment by blysik** [comment]
I think RHEL/Centos standard is to add default logrotate.d files in your RPM.  See nginx as an example.

**Comment by lomik** [comment]
Logrotate config has been added in 0.10.0 
https://github.com/lomik/go-carbon/blob/master/deploy/go-carbon.logrotate

**Comment by towolf** [comment]
@lomik 
```
logrotate -v -f /etc/logrotate.conf
reading config file /etc/logrotate.conf
including /etc/logrotate.d
...
Ignoring go-carbon because of bad file mode.
```

I fixed this by doing `chmod g-w /etc/logrotate.d/go-carbon`

**Comment by lomik** [comment]
@towolf, thanks for report.

What OS is used?

**Comment by towolf** [comment]
Ubuntu 16.04,

Strangely you set `0644` when building, but in the debian package it ends up as `0664`:

```-rw-rw-r-- 1 towolf towolf 124 Jun 21 21:07 foo/etc/logrotate.d/go-carbon```

**Comment by lomik** [comment]
I don't set anything. Actual build script here https://github.com/lomik/go-carbon/blob/master/Makefile#L80

**Comment by towolf** [comment]
I think this is the issue for that file: https://github.com/jordansissel/fpm/issues/1171

**Comment by bzed** [comment]
fpm is the worst way ever to build debian packages.

If there is enough interest I can see if we can get go-carbon into debian, properly packaged. But that would meanm that all dependencies need to be packaged, which might be some insame work

**Comment by lomik** [comment]
go-carbon was properly packaged before 0.10 release. I'm tired of supporting this.
Now travis automatically builds binary (with cross-build of 32-bit version) and rpm/deb packages using fpm.

**Comment by claytonpeters** [comment]
I've just come across this issue myself from the RPM running on Red Hat Enterprise Linux 7:

<pre># logrotate -d -v /etc/logrotate.d/go-carbon
Ignoring /etc/logrotate.d/go-carbon because of bad file mode.

# rpm -ql -v go-carbon | fgrep logrotate
-rw-rw-r--    1 root    root                      124 Jun 21 20:09 /etc/logrotate.d/go-carbon</pre>

If my understanding of FPM is correct, then:

<pre>--rpm-attr 0644,root,root:/etc/logrotate.d/go-carbon</pre>

Might do the trick.

**Comment by lomik** [comment]
I have reproduced and fixed problem in travis-ci. Next release should be ok

**Comment by lomik** [comment]
Released in 0.11.0
